model/strand.rb

```diff
@@ -73,11 +75,32 @@ def load(snap = nil)
   end
 
   def unsynchronized_run
+    stack.each do |frame|
+      next unless (deadline_at = frame["deadline_at"])
+
+      if label == frame["deadline_target"].to_s
+        frame.delete("deadline_target")
+        frame.delete("deadline_at")
```

Comment:
does it ever happen that any frame except the current one can reasonably resolve a label deadline? While we always have to check all frames to Upsert a Page+Strand, it seems like deadline deletions would be done off the top only, right?

Reply:
Yep, that is right. Moved resolution check out of the loop.